sys-utils/hwclock-rtc.c

```diff
@@ -150,15 +132,23 @@ static int open_rtc_or_exit(const struct hwclock_control *ctl)
 static int do_rtc_read_ioctl(int rtc_fd, struct tm *tm)
 {
 	int rc = -1;
-
-	rc = ioctl(rtc_fd, RTC_RD_TIME, tm);
+	struct rtc_time rtc_tm;
+	rc = ioctl(rtc_fd, RTC_RD_TIME, &rtc_tm);
```

Comment:
The same should probably done for `ioctl(RTC_SET_TIME)`.

Reply:
Agreed, thank you @t-ch for the review.